### PR TITLE
Deprecate models

### PR DESCRIPTION
**Pull request by sgugger**
# What does this PR do?

This PR creates the precedent of deprecating models in the library. By deprecating we indicate that we will stop maintaining such models, but there is no intention of actually removing those models and breaking support for them (they might one day move into a separate repo/on the Hub but we would still add the necessary imports to make sure backward compatibility stays). The main point is that we stop testing those models (to ease a bit the burden on our CI).

Deprecated models are moved in models/deprecated so direct import of objects from their modeling files will break (though that's easily fixed by adding the `.deprecated` in the path). They are removed from the `tests` folder and a mention is added in the doc page of the model.

The heuristic to pick the deprecated models in this PR is: models older than a year that got less than a cumulated 1,000 downloads (over all checkpoints) in the last 30 days (counting deduplicated downloads).

### CODE REVIEW

**Review by amyeroberts — APPROVED** [review]
Thanks for adding this deprecation logic!

**Review by ydshieh — APPROVED** [review]
Thanks for taking care of the CI cost!

File: docs/source/en/model_doc/bort.md [hunk 1]
```diff
@@ -16,6 +16,12 @@ rendered properly in your Markdown viewer.
 
 # BORT
 
+<Tip warning={true}>
+
+This model is in maintenance mode only, so we won't accept any new PRs changing its code.
```
Comment (LysandreJik) [inline docs/source/en/model_doc/bort.md:21]: I wonder if we shouldn't also mention the last version of `transformers` in which this model was supported. So something like this:

```suggestion
This model is in maintenance mode only, so we won't accept any new PRs changing its code.

If you run into any issues running this model, please reinstall the last version that supported this model: v4.30.0. You can do so by running the following command: `pip install -U transformers==4.30.0`.
```

**Review by LysandreJik — APPROVED** [review]
Yes, I think the combination of:
- Keeping the models in the main init
- Marking them as deprecated
- Not having them tested

is good.

**Comment by ydshieh** [comment]
FMI: (M=my)

I probably need to do something with daily CI to take this PR's change into account!

**Pull request merged**